src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_HUEditor_Launcher.java

```diff
@@ -59,24 +71,40 @@ protected ProcessPreconditionsResolution checkPreconditionsApplicable()
 	@Override
 	protected String doIt()
 	{
-		final IView husToPickView = createHUsToPickView();
+		final List<Integer> availableHUIdsToPick = retrieveAvailableHuIdsForCurrentShipmentScheduleId();
+
+		final IView husToPickView = createHUEditorView(availableHUIdsToPick);
 
 		getResult().setWebuiIncludedViewIdToOpen(husToPickView.getViewId().getViewId());
 		getResult().setWebuiViewProfileId("husToPick");
-
 		return MSG_OK;
 	}
 
-	private IView createHUsToPickView()
+	private List<Integer> retrieveAvailableHuIdsForCurrentShipmentScheduleId()
```

Comment:
hi @metas-rc 
FYI: i just merged my stuff from metasfresh/metasfresh#4248 where i also had to edit this laucher.

* Rather than adding filter logic to this process, please add it to `HUsToPickViewFactory.createViewRequest()`.
* `IHUPickingSlotBL` has two methods: `retrieveAvailableHUsToPick()` and `retrieveAvailableSourceHUs()`. Both categories of HUs need to be availablein the HU-editor, so it makes no sense to restrict the HUEditor to either oth them
* instead of loading and providing an explicit list of IDs, please use an IHUQueryBuilder. See `HUReservationDocumentFilterService` for how i would proposed to du it.